housekeeping/src/main/java/com/hotels/housekeeping/fs/HousekeepingFileSystem.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (C) 2016-2019 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.housekeeping.fs;
+
+import java.io.IOException;
+
+import org.apache.hadoop.fs.Path;
+
+public interface HousekeepingFileSystem {
+
+  boolean exists(Path path) throws IOException;
+
+  boolean delete(Path path, boolean b) throws IOException;
```

Comment:
Give that `b` a better name at the very least, or add some javadoc, or replace it with an enum. Uncle Bob would be crying if he saw this....

Reply:
oops